apps/directory/src/components/biobankcards-components/MatchesOn.vue

```diff
@@ -26,127 +26,158 @@ export default {
-        };
-      });
-      return filterInfoDictionary;
+      return this.filtersStore.filterFacets.reduce(
+        (
+          accum: Record<string, { column: any; label: string }>,
+          filter: {
+            facetIdentifier: string;
+            applyToColumn: any;
+            facetTitle: string;
+          }
+        ) => {
+          accum[filter.facetIdentifier] = {
+            column: filter.applyToColumn,
+            label: filter.facetTitle,
+          };
+          return accum;
+        },
+        {}
+      );
     },
-    activeFilters() {
+    activeFilters(): Record<string, any> {
       return this.filtersStore.filters;
     },
-    filterOptionsCache() {
+    filterOptionsCache(): Record<string, { value: string; text: string }[]> {
       return this.filtersStore.filterOptionsCache;
     },
     matches() {
-      let matches = [];
+      let matches: IMatch[] = [];
       const facetIdentifiers = Object.keys(this.activeFilters);
       for (const facetIdentifier of facetIdentifiers) {
         const activeFilterValues = this.activeFilters[facetIdentifier];
 
-        if (!activeFilterValues || !Array.isArray(activeFilterValues)) {
-          continue;
-        } /** no need to check further if there are no active filters */
-
-        const filterColumn = this.filterInfoDictionary[facetIdentifier].column;
-        const columns = Array.isArray(filterColumn)
-          ? filterColumn
-          : [filterColumn];
-
-        for (const column of columns) {
-          const filterLabel = this.filterInfoDictionary[facetIdentifier].label;
-          const potentialMatch = this.extractValue(column, this.viewmodel);
-
-          if (!potentialMatch) {
-            continue;
-          }
-
-          const isArray = Array.isArray(potentialMatch);
+        if (activeFilterValues?.length) {
+          /** no need to check further if there are no active filters */
 
-          const match = { name: filterLabel, value: [] };
+          const filterColumn: string | string[] =
+            this.filterInfoDictionary[facetIdentifier].column;
+          const columns: string[] = Array.isArray(filterColumn)
+            ? filterColumn
+            : [filterColumn];
 
-          for (const activeFilterValue of activeFilterValues) {
-            /** need to find the correct filter value instead of the name */
+          for (const columnId of columns) {
+            const filterLabel =
+              this.filterInfoDictionary[facetIdentifier].label;
+            const potentialMatch = extractValue(columnId, this.viewmodel);
 
-            const optionsCache = this.filterOptionsCache[facetIdentifier];
-            if (!optionsCache) {
-              continue; /** if the filteroption does not exist */
-            }
-            /** we are dealing with a multiple ontology, e.g. diagnosis available we can not deal with that yet */
-            if (!Array.isArray(optionsCache)) {
+            if (!potentialMatch) {
```

Comment:
suggest to move complex logic to separate module and add test; as documentation , and to avoid regression issues